src/list/whole/view.rs

```diff
@@ -660,43 +676,69 @@ where
+                        highlight_class
+                    )}></td>
+                    <td class="list-whole-list-flat-check list-whole-list-second-placeholder"></td>
+                    {
+                        if has_items {
+                            html! {
+                                <td colspan={cols.to_string()} class="list-whole-list-second-pages">
+                                    <div class="list-whole-list-pages-inner">
+                                        <Pages::<Self>
+                                            txt={ctx.props().txt.clone()}
+                                            language={ctx.props().language}
+                                            parent_message={Message::MovePage}
+                                            pages_info={Rc::clone(&ctx.props().pages_info)}
+                                            pages_info_cache={self.pages_info}
+                                            num_pages={DEFAULT_NUM_PAGES}
+                                        />
+                                        <div class="list-whole-list-second-add" onclick={onclick_add_second.clone()}>
+                                            { add_text }
+                                        </div>
+                                    </div>
+                                </td>
+                            }
+                        } else if cols > 1 {
+                            let button_colspan = cols - 1;
+                            html! {
+                                <>
+                                    <td class="list-whole-second-empty">
+                                        <span class="list-whole-second-empty-message">{ empty_message.clone() }</span>
+                                    </td>
+                                    <td colspan={button_colspan.to_string()} class="list-whole-second-empty-action">
+                                        <div class="list-whole-list-second-add" onclick={onclick_add_second}>
+                                            { add_text }
+                                        </div>
+                                    </td>
+                                </>
+                            }
+                        } else {
+                            html! {
+                                <td class="list-whole-second-empty">
+                                    <div class="list-whole-second-empty-row">
+                                        <span class="list-whole-second-empty-message">{ empty_message.clone() }</span>
+                                        <div class="list-whole-list-second-add" onclick={onclick_add_second}>
+                                            { add_text }
+                                        </div>
+                                    </div>
+                                </td>
+                            }
+                        }
+                    }
```

Comment:
Wouldn’t it be enough to base the logic only on whether has_items is true? Was there a specific reason for adding the extra condition?